<a href="https://colab.research.google.com/github/Pradip240/IPL-T20-Cricket-Analysis/blob/main/Ameen_IPL_T20_Cricket_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Indian Premier League(IPL) is a professional Twenty20 cricket league in India contested during March or April and May of every year by eight teams representing eight different cities in India. The league was founded by the Board of Control for Cricket in India(BCCI) in 2008. </b>

## <b> Explore and analyze the data to discover results and statistics for different teams playing in IPL. </b>

##Let's load and explore the dataset

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
#Importing the Numpy & the Pandas libraries.
import numpy as np
import pandas as pd

In [6]:
file_path = '/content/drive/MyDrive/AlmaBetter Data Science/1. PYTHON/CAPSTONE PROJECT/IPL Data/'
ipl_df = pd.read_csv(file_path + 'matches.csv')
ipl_df1 = pd.read_excel(file_path + 'Players.xlsx')
ipl_df2 = pd.read_csv(file_path + 'deliveries.csv')
ipl_df3 = pd.read_csv(file_path + 'teams.csv')
ipl_df4 = pd.read_csv(file_path + 'most_runs_average_strikerate.csv')
ipl_df5 = pd.read_csv(file_path + 'teamwise_home_and_away.csv')

In [9]:
#Checking the shape multiple datasets
df_ipl=[ipl_df.shape, ipl_df1.shape, ipl_df2.shape, ipl_df3.shape, ipl_df4.shape, ipl_df5.shape]
df_ipl


[(756, 18), (566, 5), (179078, 21), (15, 1), (516, 6), (14, 7)]

In [ ]:
#Checking the first five observations of matches dataset
matches=ipl_df.head()
matches

In [ ]:
#Checking the first five observations of players dataset
players = ipl_df1.head()
players

In [ ]:
ipl_df1.info()

In [ ]:
#Checking the first five observations of deliveries dataset
deliveries = ipl_df2.head()
deliveries

In [ ]:
ipl_df2.info()

In [ ]:
#Checking the first ten observations of teams dataset
teams = ipl_df3.head(10)
teams

In [ ]:
ipl_df3.info()

In [ ]:
#Checking the first ten observations of most_runs_average_strikerate dataset
most_runs_average_strikerate = ipl_df4.head(10)
most_runs_average_strikerate

In [ ]:
ipl_df4.info()

In [ ]:
#Checking the first ten observations of teamwise_home_and_away dataset
teamwise_home_and_away = ipl_df5.head(10)
teamwise_home_and_away

In [ ]:
ipl_df5.info()

In [ ]:
#Checking the last five observations of matches dataset
ipl_df.tail()

In [ ]:
#Checking for the null values in the matches dataset
ipl_df.isnull()

In [ ]:
#Checking the info of the matches datset
ipl_df.info()


In [ ]:
#suming up together the null values for each column from matches dataset
ipl_df.isnull().sum()

In [28]:
#Looking up for the columns in the matches dataset
ipl_df.columns

Index(['id', 'Season', 'city', 'date', 'team1', 'team2', 'toss_winner',
       'toss_decision', 'result', 'dl_applied', 'winner', 'win_by_runs',
       'win_by_wickets', 'player_of_match', 'venue', 'umpire1', 'umpire2'],
      dtype='object')

In [ ]:
#Getting the unique values from dates of matches dataset.
ipl_df['date'].unique()

# So what can we expect from the multiple IPL datasets?
##The datasets are:
###1.matches.csv
###2.Players.xlsx
###3.deliveries.csv
###4.teams.csv
###5.most_runs_average_strikerate.csv
###6.teamwise_home_and_away.csv

###In order to understand our data, we can look at each variable and try to understand their meaning and relevance to this problem.
###The datsets consists of multiple observations & columns. Let's go ahead & get familiar with the columns.

##From matches dataset we have:-
###*id: Basically the IPL match ID.
###*Season: The IPL season, which in our dataset if from season 2008 - season 2019
###*City: The cities where the IPL matches were played.
###*Date: The date on which the matches were played or held.
###*Team1 & Team2: The teams participated in the IPL matches since 2008.
###*Toss_winner: The team that won the toss.
###*Toss_decision: The decision taken by the toss winning team either to 'Bat' or 'Field' first.
###*Result: The result column consists of (normal, tie, no result) of the match.
###Dl_applied: The column consists the numerical values (0 or 1) which indicates whether the Duckworth-Lewis method was applied or not.
###*Winner: Winner of the respective match.
###*Win_by_runs: The runs by which the team batting first won.
###*Win_by_wickets: The wickets by which the team batting second won.
###*Player_of_the_match: The player who performed exceptionally good in the match.
###*Venue: The venue where the match was hosted.
###*Umpire1 & Umpire2: Two on-field umpires during the match.
###*Umpire3: Basically the third Umpire.

##From the players dataset:-(rows 556, columns 5)
###*It consists of following columns- player_name, DOB, Batting_skill, Bowling_skill & Country

##From the deliveries dataset:-(rows 179078, columns 21)
###*It consists of following columns- match_id, inning, batting_team,	bowling_team,	over,	ball,	batsman,	non_striker,	bowler,	is_super_over,	wide_runs,	bye_runs,	legbye_runs,	noball_runs,	penalty_runs,	batsman_runs,	extra_runs,	total_runs,	player_dismissed,	dismissal_kind,	fielder.

##From teams dataset:- (rows 15, column 1)
###*Just has the team names in it.

##From most_runs_average_strikerate dataset:- (rows 516, columns 6)
###*It consists of following columns- batsman, total_runs, out,	numberofballs, average, strikerate.

##From teamwise_home_and_away datset:- (rows 14, columns 7)
###*It consists of following columns- team, home_wins, away_wins, home_matches	away_matches,	home_win_percentage,	away_win_percentage.

##Observations from the dataset:
##1.The matches dataset has 756 rows and 18 columns.
##2.The matches dataset hardly has any null values apart from column 'umpire3' which has significant null values. (i.e 637 null values)
##3.So we can drop the 'umpire3' column from the dataframe and proceed further with the EDA. 

In [26]:
#Droping the 'Umpire3' column from the dataset
ipl_df = ipl_df.drop(columns=['umpire3'], axis=1)

In [ ]:
ipl_df.head()

##As our dataset has been cleaned, we can now do the in-depth analysis and visualization.